### ISSUE TITLE
Current directory inconsistency on Mac and Windows

### ISSUE THREAD

**Issue by nicolas-p** [factor/factor#2229]
`"." directory-entries .` gives me the contents of the filesystem root (/) on macOS but gives me the contents of the Factor directory on Windows.

Similarly, if I have a "test.txt" file in the Factor directory (or the same directory as the deployed application), I can read it with `"test.txt" utf8 file-lines` on Windows, but on macOS I need to enter the absolute path.

Could the Mac version behave like the Windows version?

**Comment by mrjbq7** [comment]
We can probably change the default current directory to be consistent, but assumptions like that in your code are fragile. 

“resource:test.txt” utf8 file-lines

Will be robust because it specifies to use the Factor root.

**Comment by catb0t** [comment]
Why does it behave differently? Two translations are missing  on macOS, `test.txt` is not being translated to `./test.txt`, and `.` is not being translated to `realpath .`. I don't have a mac, does `"./"` or `"./file"` give different results than `"."` and `"file"`? 


Is this because `directory-entries` macOS backend calls Darwin's `stat` with a a literal `"."` argument, or is it because the Windows and Linux backends first do `realpath .` before `stat(2)` etc?

**Comment by nicolas-p** [comment]
I can confirm that `"."` and `"./"` give the same result.
```
"./" directory-entries "." directory-entries = .
t
```

**Comment by mrjbq7** [comment]
It is simply that “.” Is relative to the current-directory. They have different current directories. It has nothing to do with real path I believe.

**Comment by catb0t** [comment]
Oh, it's only about starting directory, otherwise they're the same, I see.

**Comment by mrjbq7** [comment]
Just to follow up, we set the default ``current-directory`` here:

https://github.com/factor/factor/blob/master/core/io/files/files.factor#L93

The problem is that when launching a macOS application from the Finder or with ``open`` on the command line, the current working directory is defaulted to root (``/``).

We could just "hard code" the ``current-directory`` to ``resource:``, but perhaps there's a better way in the Factor binary to indicate to macOS we don't want that behavior.

**Comment by mrjbq7** [comment]
Possible patch, although I'd want to make sure this works for deployed binaries:

```diff
diff --git a/core/io/files/files.factor b/core/io/files/files.factor
index 6725481cb1..2ce975647d 100644
--- a/core/io/files/files.factor
+++ b/core/io/files/files.factor
@@ -90,8 +90,8 @@ PRIVATE>
     [ image-path parent-directory ] unless* "resource-path" set-global ;
 
 [
-    cwd current-directory set-global
     OBJ-IMAGE special-object alien>native-string \ image-path set-global
     OBJ-EXECUTABLE special-object alien>native-string \ vm-path set-global
     init-resource-path
+    "resource-path" get-global current-directory set-global
 ] "io.files" add-startup-hook
```

**Comment by jonenst** [comment]
This is wrong for deployed binaries (for example an 'ls' written in factor)

Imho when launching a program not from the terminal, it is normal that the
cwd is not set.

**Comment by nomennescio** [comment]
"Current-directory" is a notoriously tricky concept to get right; typically it entirely dependent on the environment.

**Comment by mrjbq7** [comment]
Fixed in 738552a909e3ad99ad88fe467b882445c44befea.
